src/game/SpellEffects.cpp

```diff
@@ -5457,6 +5457,33 @@ void Spell::EffectScriptEffect(SpellEffectIndex eff_idx)
                     unitTarget->RemoveAurasAtMechanicImmunity(IMMUNE_TO_ROOT_AND_SNARE_MASK, 30918, true);
                     break;
                 }
+                
+                
+                case 32027:				// Expedition Flare
+			{
+				if (!unitTarget)
+					return;
+
+				uint32 creature_id;
+				uint32 rand = urand(0, 99);
+
+				//summon Expedition Preserver or Expedition Scout 
+				if (rand < 50)
+					creature_id = 18194;
+				else 
+					creature_id = 18126;
+		
+				float angle = unitTarget->GetOrientation();
+			
+				Creature* pCreature = m_caster->SummonCreature(creature_id, 0.0f, 0.0f, 0.0f, angle, TEMPSUMMON_TIMED_OOC_OR_DEAD_DESPAWN, 60000, true);
+
+				if (pCreature->AI())
+					pCreature->AI()->AttackStart(m_caster->getVictim());
+				return;
+
+			}
+                
+        
```

Comment:
Please use the code style guidelines https://github.com/cmangos/issues/wiki/Coding-Standards

Also, don't summon creature as ActiveObject (remove the last summon parameter). and maybe also check for ```m_caster->getVictim())``` in case there is no victim.

Finally no [Hungarian notation](https://msdn.microsoft.com/en-us/library/windows/desktop/aa378932%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396) for variables in SpellEffects (don't ask me why, but this is how everything was agreed before)